### PR TITLE
Validate label smoothing range in loss functions

### PR DESCRIPTION
**Pull request by kaivalya-cyber**
Fixes tensorflow/tensorflow#122717

Port the label_smoothing validation from the deprecated TensorFlow Keras
implementation to Keras 3.

Functional and class-based categorical_crossentropy and
binary_crossentropy now raise ValueError when a numeric label_smoothing
value is outside [0, 1]. Added regression tests for both APIs.

Validation completed:
- Ruff passes
- Python syntax compilation passes
- Diff checks pass
- Focused pytest is blocked locally because optree is not installed

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by kaivalya-cyber** [comment]
- NumPy scalar values are now validated through float conversion.
- Boolean values are explicitly rejected.
- Tensor values are safely skipped during Python-side validation.
- Added regression coverage for invalid booleans and NumPy scalar values.

**Comment by kaivalya-cyber** [comment]
Thanks for the review. Updated the validation to accept NumPy real scalar values while rejecting Python and NumPy booleans, and preserved the dynamic Tensor path. Added regression coverage for these cases and pushed the update in 704983ecb. Ruff, formatting, compilation, and diff checks pass locally; the focused pytest was unavailable because optree is not installed.

File: keras/src/losses/losses.py [hunk 1]
```diff
@@ -2184,6 +2187,21 @@ def categorical_crossentropy(
...
+            raise ValueError(
+                "`label_smoothing` must be a float or int. "
+                f"Received: label_smoothing={label_smoothing}"
+            )
+        value = float(label_smoothing)
+        if not 0 <= value <= 1:
+            raise ValueError(
+                "`label_smoothing` must be in the range [0, 1]. "
+                f"Received: label_smoothing={label_smoothing}"
+            )
```
Comment (gaga1313) [inline keras/src/losses/losses.py]: Can we create a helper function for this. This code is replicated twice. Thanks

**Review by gaga1313 — COMMENTED** [review]
Thanks for the fix! One last change.

**Comment by kaivalya-cyber** [comment]
@gaga1313 Done. Extracted the duplicated validation into a _validate_label_smoothing helper — both categorical_crossentropy and binary_crossentropy now call it, with the checks and error messages unchanged. Pushed in 2c53f29d9; Ruff, formatting, compilation, and diff checks pass locally, and the focused pytest runs in CI.

**Comment by gaga1313** [comment]
@kaivalya-cyber  Thanks for the fix! @hertschuh the PR looks good to me!
